src/interpolation/utils.jl

```diff
@@ -17,10 +33,16 @@ function compute_bowyer_envelope!(envelope, tri::Triangulation, history::Inserti
     v = i == I(n + 1) ? j : i # get a vertex on the envelope
     push!(envelope, v)
     for i in 2:num_triangles(all_triangles)
-        v = get_adjacent(temp_adjacent, I(n + 1), v)
+        #=
+        Get the vertex associated with the edge `(point, points[v])` such that the 
+        returned point (the new `v`) forms a positively oriented triangle
+        with that edge.  Then, add the newly found `v` to the envelope.
+        To be clear, `n+1` is the index of the newly inserted vertex in the triangulation.
+        =#
+        v = get_adjacent(temp_adjacent, I(n + 1), v) 
         push!(envelope, v)
     end
```

Comment:
I'm not 100% sure what this loop is doing - is there some guarantee that all triangles added have `n+1` as a vertex, and we're just getting each individual natural neighbour from this loop?

Reply:
That loop is building up the "envelope". All triangles will have `n+1` as a vertex. I'll draw a picture

Reply:
Gotcha - that's what I thought was happening but just wanted to confirm.  Thanks!

Reply:
![image](https://github.com/user-attachments/assets/01834e60-767a-45f5-8850-d98c4ec41e68)

By pivoting around `n+1` (which I've called `i+1` in the image for some reason), we can traverse the boundary of the envelope counter-clockwise by looping over the triangles. The orientation is important since we compute areas later